src/toolchains.ts

```diff
@@ -0,0 +1,143 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+import * as constants from './constants';
+
+import { getBooleanInput } from './util';
+import { create as xmlCreate } from 'xmlbuilder2';
+
+export async function configureToolchains(
+  version: string,
+  distributionName: string,
+  jdkHome: string,
+  toolchainId: string | undefined
```

Comment:
```suggestion
  toolchainId?: string
```

TypeScript has sugar for optional parameters

Reply:
Fixed